src/server/game/Server/Protocol/Opcodes.cpp

```diff
@@ -960,7 +960,7 @@ void OpcodeTable::Initialize()
     /*0x33D*/ DEFINE_SERVER_OPCODE_HANDLER(SMSG_MOTD,                      STATUS_NEVER);
     /*0x33E*/ DEFINE_SERVER_OPCODE_HANDLER(SMSG_MOVE_SET_CAN_TRANSITION_BETWEEN_SWIM_AND_FLY, STATUS_NEVER);
     /*0x33F*/ DEFINE_SERVER_OPCODE_HANDLER(SMSG_MOVE_UNSET_CAN_TRANSITION_BETWEEN_SWIM_AND_FLY, STATUS_NEVER);
-    /*0x340*/ DEFINE_HANDLER(CMSG_MOVE_SET_CAN_TRANSITION_BETWEEN_SWIM_AND_FLY_ACK, STATUS_NEVER, PROCESS_INPLACE,  &WorldSession::Handle_NULL                     );
+    /*0x340*/ DEFINE_HANDLER(CMSG_MOVE_SET_CAN_TRANSITION_BETWEEN_SWIM_AND_FLY_ACK, STATUS_LOGGEDIN, PROCESS_THREADSAFE, &WorldSession::HandleMoveSetCanTransitionBetweenSwinAndFlyAck);
```

Comment:
why does it qualify as threadsafe? honestly don't know myself

Reply:
it's like the other movement packet handlers: a thread safe message is a message which can be handled in one of the Map update threads while other Map updates are running concurrently. Basically, everything that only concern things happen in one map (for example dealing damage to a unit or movement) is thread safe. Things that cross map boundaries (chat, mail, LFG, ...) are non thread safe.